src/main/java/com/cloudbees/plugins/credentials/CredentialsNameProvider.java

```diff
@@ -89,13 +91,14 @@ Result name(@NonNull Credentials credentials, @NonNull Class<?> clazz) {
         NameWith nameWith = clazz.getAnnotation(NameWith.class);
         if (nameWith != null) {
             try {
-                CredentialsNameProvider nameProvider = nameWith.value().newInstance();
+                CredentialsNameProvider nameProvider = nameWith.value().getDeclaredConstructor().newInstance();
```

Comment:
```suggestion
                CredentialsNameProvider nameProvider = nameWith.value().getConstructor().newInstance();
```

Reply:
This was not resolved as the `getDeclaredContructor` is still proposed.

Reply:
Sorry, seems I've missed that.

> ⁉️ Not sure to understand the improvement here.

`newInstance()` is deprecated and `getDeclaredConstructor().newInstance()` is suggested by the docs.